src/components/IncorporationAgreement/AgreementType.vue

```diff
@@ -0,0 +1,150 @@
+<template>
+  <div>
+    <div id="agreement-summary" v-if="isSummary">
+      <v-card flat>
+        <!-- Summary Header -->
+        <div class="agreement-summary-header">
+            <v-icon>mdi-handshake</v-icon>
+            <label class="agreement-summary-title"><strong>Incorporation Agreement and Benefit </strong></label>
+        </div>
+
+        <!-- Summary Warning -->
+        <div v-if="showErrorSummary" class="agreement-invalid-message">
+            <span>
+            <v-icon color="#1976d2">mdi-information-outline</v-icon>
+            This step is not complete.
+            <router-link id="router-link" :to="{ path: '/incorporation-agreement' }">
+                Return to this step to complete it.
+            </router-link>
+            </span>
+        </div>
+        <!-- Summary Content -->
+        <div v-else class="summary-desc">
+            <div><v-icon color="green" class="agreement-valid-icon">mdi-check</v-icon></div>
+             <div>
+                {{ selectedAgreementDescription }}
+            </div>
+        </div>
+      </v-card>
+    </div>
+    <div v-else style="margin-left:1rem;">
+      <v-card flat>
+        <v-radio-group v-model="agreementType" @change="changeAgreementType" class="agreement-option-list">
+          <v-radio v-for="(item, index) in incorporationAgreementTypes"
+           :key="index" :value="item.code" :id="`agreement-type-${item.code}`">
+            <template slot="label">
+              <div v-html="item.description" class="agreement-option"/>
+            </template>
+          </v-radio>
+        </v-radio-group>
+      </v-card>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+// Libraries
+import { Component, Vue, Prop } from 'vue-property-decorator'
+import { Action, State } from 'vuex-class'
+
+// Interfaces
+import { ActionBindingIF } from '@/interfaces'
+import { AgreementTypeResource } from '@/resources'
+
+@Component
+export default class AgreementType extends Vue {
+   private incorporationAgreementTypes = AgreementTypeResource
+
+  // State
+  @State(state => state.stateModel.incorporationAgreementStep.agreementType)
+  readonly agreementTypeState: string | null
```

Comment:
This looks strange...
On line 56, types = resource
On this line, state = type

Would it make sense to rename some of these so they agree with each other?

Reply:
Also on line 78  below.

Reply:
Not sure what you meant here. I have renamed it to incorporationAgreementTypeResource

Reply:
Thanks for fixing that one.

Here's an example that I found confusing:
```
@State(state => state.stateModel.incorporationAgreementStep.agreementType)
  readonly agreementTypeState: string | null
```
The property in state is "agreementType" but in this component you rename it "agreementTypeState".

Why not map it with the same name?
```
@State(state => state.stateModel.incorporationAgreementStep.agreementType)
  readonly agreementType: string | null
```

Reply:
OK, I completely missed that you have 2 things you want to name the same (the global and the local), so you used extended names to differentiate them. (I was confused.)

What you have now is fine.